src/systems/sensors/Sensors.cc

```diff
@@ -417,6 +417,7 @@ void Sensors::Update(const UpdateInfo &_info,
                      EntityComponentManager &_ecm)
 {
   IGN_PROFILE("Sensors::Update");
+  std::unique_lock<std::mutex> lock(this->dataPtr->renderMutex);
```

Comment:
I know this is merged, but I just noticed this. If I understand correctly, having this mutex lock here could prevent sensors from updating in parallel with physics. This would happen if the `Sensors` system comes before the `Physics` system in the SDFormat file. Here's the sequence of events that could happen:
1. In the `Sensors::PostUpdate`, a new render is triggered that takes a long time to complete. The render thread locks `renderMutex` and starts the render.
2. In the next phase, `Sensors::Update` tries to acquire `renderMutex` and blocks because the render is not complete. This also blocks `Physics::Update` from being called because it comes after `Sensors::Update`.

Can some one verify this?